src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,112 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status    - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
+                                                               final List<Long> userIds) {
+
+    try {
+      RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
+      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
+        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
+            .toResponse();
+      }
+
+      Map<String, List<Long>> failedUpdates = new HashMap<>();
+
+      for (Long userId : userIds) {
+        modifyTeacherPendingStatusForUser(userId, requestingUser, status, failedUpdates);
+      }
+
+      if (!failedUpdates.isEmpty()) {
+        String errorMessage = "";
+
+        if (failedUpdates.containsKey(USERS_NOT_FOUND)) {
+          errorMessage = String.format("One or more users could not be found: %s. ",
+              failedUpdates.get(USERS_NOT_FOUND));
+        }
+
+        if (failedUpdates.containsKey(FAILED_TO_SEND)) {
+          errorMessage += String.format("Emails could not be sent to userIds: %s",
+              failedUpdates.get(FAILED_TO_SEND));
+        }
+        return new SegueErrorResponse(Status.BAD_REQUEST, errorMessage).toResponse();
+      }
+
+    } catch (NoUserLoggedInException e) {
```

Comment:
You're surrounding the whole method between `try`, when in reality only a handful of lines need to be surrounded by it. Given that all you're doing when caching these exceptions is to return an error response, it would be better to only surround the parts of the code that can cause these exceptions.

In the case of `NoUserLoggedInException`, I think you only need it in the first block:

```java
      RegisteredUserDTO requestingUser = userManager.getCurrentRegisteredUser(request);
      if (!isUserAnAdminOrEventManager(userManager, requestingUser)) {
        return new SegueErrorResponse(Status.FORBIDDEN, ACCESS_DENIED_MESSAGE)
            .toResponse();
      }
```

In the case of `SegueDatabaseException`, it only happens in the `for` loop.